src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -375,20 +379,37 @@ public void onRandomMapButtonClicked() {
     mapListView.scrollTo(mapIndex);
   }
 
-  public void onGenerateMapButtonClicked() {
+  public void onGenerateMapButtonClicked(){
+    onGenerateMap();
+  }
+
+  private void onGenerateMap() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
-        Platform.runLater(() -> {
-          initMapSelection();
-          mapListView.getItems().stream()
-              .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
-              .findAny().ifPresent(mapBean -> {
-            mapListView.getSelectionModel().select(mapBean);
-            mapListView.scrollTo(mapBean);
-            setSelectedMap(mapBean);
+      String generatorVersion = mapGeneratorService.queryMaxSupportedVersion();
+      String majorVersion = generatorVersion.split("\\.")[0];
```

Comment:
I think we have a class that can parse semantic versions... If we do, which I think use that

Reply:
Used ComparableVersion